newrelic-agent/src/main/java/com/newrelic/agent/sql/DefaultSlowQueryListener.java

```diff
@@ -43,7 +43,7 @@ public DefaultSlowQueryListener(String appName, double thresholdInMillis) {
     @Override
     public <T> void noticeTracer(Tracer tracer, SlowQueryDatastoreParameters<T> slowQueryDatastoreParameters) {
         if (tracer.getDurationInMilliseconds() > thresholdInMillis) {
-            String query = (String) slowQueryDatastoreParameters.getQuery();
+            String query = slowQueryDatastoreParameters.getQuery().toString();
```

Comment:
This fixed a ClassCastException, but it is an oversimplification that may not work for all query objects from different databases.
Ideally here it should use the queryConverter in the SlowQueryDatastoreParameters instead of assuming the toString call will return a query.